x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/TextExpansionQueryBuilder.java

```diff
@@ -51,6 +50,43 @@ public class TextExpansionQueryBuilder extends AbstractQueryBuilder<TextExpansio
     private SetOnce<TextExpansionResults> weightedTokensSupplier;
     private final TokenPruningConfig tokenPruningConfig;
 
+    public enum AllowedFieldType {
+        RANK_FEATURES("rank_features"),
+        SPARSE_VECTOR("sparse_vector");
+
+        private final String typeName;
+
+        AllowedFieldType(String typeName) {
+            this.typeName = typeName;
+        }
+
+        public String getTypeName() {
+            return typeName;
+        }
+
+        public static boolean isFieldTypeAllowed(String typeName) {
+            for (AllowedFieldType fieldType : values()) {
+                if (fieldType.getTypeName().equals(typeName)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public static String getAllowedFieldTypesAsString() {
+            StringBuilder result = new StringBuilder();
+            for (AllowedFieldType fieldType : values()) {
+                result.append(fieldType.getTypeName()).append(", ");
+            }
+
+            // Remove the trailing ", "
+            if (result.length() > 0) {
+                result.setLength(result.length() - 2);
+            }
+            return result.toString();
```

Comment:
You can clean this up a bit. 

```suggestion
            return Arrays.stream(values()).map(value -> value.typeName).collect(Collectors.joining(", "));
```

Reply:
Updated.